src/Transformers/RegexFilter.php

```diff
@@ -122,6 +122,10 @@ public function compatibility() : array
      */
     public function transform(array &$samples) : void
     {
+        if (empty($this->patterns)) {
```

Comment:
If there are not patterns, then this method is a no-op: No need to iterate over the entire dataset and type checks to perform nothing.